### PR TITLE
Feature jsonp highlighter

### PR DESCRIPTION
**Pull request by yidea**
Changes: 
- add jsonp support
- add highlighter 
- add triggerHandler in eventbus
- add jasmine unit testing specs for new features 
- fix some code formatting (padding)
- add playground2.html for testing with walmart.com typeahead service 
- update jquery lib to 10.2 to comply with Atlas env, added handlerbars.js for templating usage and underscore,  update grunt-contrib-jasmine lib to 0.5

@ryan-roemer @alexlande

### CODE REVIEW

**Comment by ryan-roemer** [comment]
@alexlande -- Can you help out and review this PR?

File: src/typeahead_view.js [hunk 1]
```diff
@@ -178,7 +178,11 @@ var TypeaheadView = (function() {
     _setInputValueToSuggestionUnderCursor: function(e) {
       var suggestion = e.data;
 
-      this.inputView.setInputValue(suggestion.value, true);
+//      this.eventBus.trigger('cursorMoved', suggestion);
```
Comment (alexlande) [inline src/typeahead_view.js]: Should this be commented out, or removed entirely?

File: test/playground2.html [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+      remote:    {
+        url:           "http://search-cdn.walmart.com/typeahead/v2/142487e83d0c/0/",
+        replace:       function (url, uriEncodedQuery) {
+          return url + _encodeQuery(uriEncodedQuery) + ".js?";
+        },
+        dataType:      "jsonp",
+        jsonpCallback: "typeaheadResult",
+        cache:         true,
+        timeout:       4000, //don't timeout for now to avoid jq typeError issue
+//        rateLimitWait: 200,
```
Comment (alexlande) [inline test/playground2.html]: Why are these options commented out?

File: test/playground3.html [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        <div class="tt-suggestion"><p>iPhone in Video Games</p></div>
+        <div class="tt-suggestion"><p>iphone 4</p></div>
+        <!--<div class="tt-suggestion"><p>iphone 5</p></div>-->
+        <!--<div class="tt-suggestion"><p>iphone 5s</p></div>-->
+      </span>
+    </div>
+  </div>
+
+  <script>
+//    new Highlighter({el: $(".tt-suggestions"), pattern: "iphone"});
```
Comment (alexlande) [inline test/playground3.html]: Commented out code again.

**Comment by alexlande** [comment]
Generally looks good except for a few extraneous commented out lines of code. Test suite is running well. There are a number of test failures in Firefox, but those are also present in master and aren't related to this PR.

File: test/playground2.html (hunk 2 above)
Comment (yidea) [inline test/playground2.html]: There are some options to play with for the prototype, for a better ux ( e.g. debounce time). I will comment those out

**Comment by ryan-roemer** [comment]
@yidea -- Please fix up code per @alexlande 's comments. Rest LGTM. Approved. Please merge (and if you don't have creds just let me know and I'll merge).

**Comment by yidea** [comment]
@ryan-roemer fixed code based on review, couldn't merge from my side, didn't find the Merge GUI. Can you help me on merging this?

**Comment by ryan-roemer** [comment]
@yidea -- You are merged, sir. ;)

**Pull request merged**